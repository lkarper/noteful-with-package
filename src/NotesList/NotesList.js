import React from 'react';
import { Link } from 'react-router-dom';
import ShortNote from '../ShortNote/ShortNote';
import NotesContext from '../NotesContext/NotesContext';

const NotesList = () => {
    return (
        <NotesContext.Consumer>
            {value => {
                const error = value.noteError;
                const errorHTML = (
                    <li className="notes-load-error">
                        <h2>Sorry, could not fetch notes from server: {error}.</h2>
                        <p>Check your connection and reload the page.</p>
                    </li>
                );

                const notes = value.notes.map(note => 
                    <ShortNote key={note.id} note={note} />
                );

                return (
                    <section className="notes">
                        {!error && !value.notes.length ? <h2>Loading...</h2> :
                            <>
                                <ul>
                                    {notes.length && !error ? notes : <li>No notes saved.</li>}
                                    {error ? errorHTML : ''}
                                </ul>
                                <Link 
                                    className="add-note-button"
                                    to={{
                                        pathname: "/add-note",
                                        state: {
                                            folderId: null,
                                        },
                                    }}
                                >
                                    Add note
                                </Link>
                            </>
                        }
                    </section>
                );
            }}
        </NotesContext.Consumer>
    );
}

export default NotesList;