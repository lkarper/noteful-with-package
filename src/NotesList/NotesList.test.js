import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NotesList from './NotesList';

describe('NotesList component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <NotesList />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NotesList />
            </BrowserRouter>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it('renders an error message when there is a notes error passed in context', () => {
        const wrapper = mount(
            <BrowserRouter>
                <NotesList />
            </BrowserRouter>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    })
});