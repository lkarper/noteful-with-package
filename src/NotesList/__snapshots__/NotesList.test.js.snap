// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList component renders an error message when there is a notes error passed in context 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesList>
      <section
        className="notes"
      >
        <h2>
          Loading...
        </h2>
      </section>
    </NotesList>
  </Router>
</BrowserRouter>
`;

exports[`NotesList component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NotesList>
      <section
        className="notes"
      >
        <h2>
          Loading...
        </h2>
      </section>
    </NotesList>
  </Router>
</BrowserRouter>
`;
